tools/common/Optimizations.cs

```diff
@@ -0,0 +1,68 @@
+namespace Xamarin.Bundler {
+	public class Optimizations
+	{
+		public const string REMOVE_UITHREAD_CHECKS = "remove-uithread-checks";
+		public bool? RemoveUIThreadChecks;
+		public bool? InlineSetupBlock;
+		public bool? InlineIsDirectBinding;
+		public bool? RemoveDynamicRegistrar;
+		public bool? InlineIntPtrSize;
+
+		public void Parse (params string [] options)
+		{
+			foreach (var option in options)
+				Parse (option);
+		}
+
+		public void Parse (string option)
+		{
+			if (option == null || option.Length < 2)
+				throw ErrorHelper.CreateError (9999, "?");
+
+			bool enabled;
+			string opt;
+			switch (option [0]) {
+			case '-':
+				enabled = false;
+				opt = option.Substring (1);
+				break;
+			case '+':
+				enabled = true;
+				opt = option.Substring (1);
+				break;
+			default:
+				opt = option;
+				enabled = true;
+				break;
+			}
+
+			switch (opt) {
+			case "remove-dynamic-registrar":
```

Comment:
👍 on infrastructure to enable/disable optimizations. Will make it easier when customers run into issues to say "try turning these on without this to see if..."